### PR TITLE
Adding continents support

### PR DESCRIPTION
**Pull request by leomanne**
Adding Continents Support

### CODE REVIEW

**Comment by leomanne** [comment]
Adding Continents support

File: Configuration.cpp [hunk 1]
```diff
@@ -123,6 +122,31 @@ bool Configuration::readConfigFile(char *path) {
...
+      } else {
+        for(Json::Value::ArrayIndex i = 0; i != root["continents"]["blacklist"].size(); i++) {
+  	std::string continent = root["continents"]["blacklist"][i].asString();
+    continents[country2u16((char*)continent.c_str())] = MARKER_DROP;
+  	trace->traceEvent(TRACE_INFO, "Adding %s to continents blacklist", continent.c_str());
+
+        }
+      }
+    }
+
```
Comment (lucaderi) [inline Configuration.cpp:149]: Refactor code in order to merge country and continent logic and remove code duplication

File: Configuration.cpp [hunk 2]
```diff
@@ -139,3 +163,12 @@ Marker Configuration::getCountryMarker(char *country) {
...
 /* ******************************************************* */
+Marker Configuration::getContinentMarker(char *continent) {
+  u_int16_t id = country2u16(continent);
+  std::unordered_map<u_int16_t, Marker>::iterator it = continents.find(id);
+
+  if(it == continents.end())
+    return(default_marker); /* Not found */
+  else
+    return(it->second);
+}
```
Comment (lucaderi) [inline Configuration.cpp:174]: Refactor here too

File: NwInterface.cpp [hunk 3]
```diff
@@ -230,7 +230,11 @@ Marker NwInterface::makeVerdict(u_int8_t proto, u_int16_t vlanId,
   const char *proto_name = getProtoName(proto);
   bool pass_local = true;
   Marker m, src_maker, dst_marker;
-  
+  // declaration of continent and Marker for continent
+  char continent[14];
```
Comment (lucaderi) [inline NwInterface.cpp:234]: Why 14 chars? Can you please explain what are the continent acronyms supported?

**Review by lucaderi — CHANGES_REQUESTED** [review]
The code is in good shape but you should not commit binary or generated files. Please remove them. When you send a PR you need to specify what have you changes and not re-upload the whole project to github

**Comment by lucaderi** [comment]
Closing PR as of https://github.com/ntop/ipt_geofence/pull/11

**Pull request closed without merging**